template_loader.go

```diff
@@ -211,3 +212,56 @@ func (h *HttpFilesystemLoader) Get(path string) (io.Reader, error) {
 
 	return h.fs.Open(fullPath)
 }
+
+// EmbededLoader support loading templates from embed.FS
+// embed.FS file names pattern are limited check https://pkg.go.dev/embed#hdr-Directives
+// and https://github.com/golang/go/issues/43854
+type EmbededLoader struct {
```

Comment:
I'm excited to see progress on this PR! Small nit: `EmbededLoader` should be `EmbeddedLoader`.